processor/src/tests/u32_ops.rs

```diff
@@ -163,63 +163,650 @@ fn u32split() {
+    let c = rand_value::<u64>();
+    test_execution(asm_op, &[b as u64, a as u64, c], &[expected as u64, c]);
+}
+
+#[test]
+fn u32div_fail() {
+    let asm_op = "u32div";
+
+    // should fail if a >= 2^32
+    test_execution_failure(asm_op, &[1, U32_BOUND], "FailedAssertion");
+
+    // should fail if b >= 2^32
+    test_execution_failure(asm_op, &[U32_BOUND, 1], "FailedAssertion");
+}
+
+#[test]
+#[should_panic = "divide by zero"]
+fn u32div_panic() {
+    let script = compile("begin u32div end");
+    let inputs = build_inputs(&[0, 1]);
+
+    // should panic if b = 0
+    execute(&script, &inputs).unwrap();
+}
+
+#[test]
+// issue: https://github.com/maticnetwork/miden/issues/94
+fn u32div_b() {
+    let build_asm_op = |param: u32| format!("u32div.{}", param);
+
+    // --- simple cases ---------------------------------------------------------------------------
+    test_execution(build_asm_op(1).as_str(), &[0], &[0]);
+    // division with no remainder
+    test_execution(build_asm_op(1).as_str(), &[2], &[2]);
+    // division with remainder
+    test_execution(build_asm_op(2).as_str(), &[1], &[0]);
+
+    // --- random u32 values ----------------------------------------------------------------------
+    let a = rand_value::<u64>() as u32;
+    let b = rand_value::<u64>() as u32;
+    let expected = a / b;
+    test_execution(build_asm_op(b).as_str(), &[a as u64], &[expected as u64]);
+
+    // --- test that the rest of the stack isn't affected -----------------------------------------
+    let c = rand_value::<u64>();
+    test_execution(
+        build_asm_op(b).as_str(),
+        &[a as u64, c],
+        &[expected as u64, c],
+    );
+}
+
+#[test]
+// issue: https://github.com/maticnetwork/miden/issues/94
+fn u32div_b_fail() {
+    let build_asm_op = |param: u64| format!("u32div.{}", param);
+
+    // should fail during execution if a >= 2^32
+    test_execution_failure(build_asm_op(1).as_str(), &[U32_BOUND], "FailedAssertion");
+
+    // should fail during compilation if b >= 2^32
+    test_param_out_of_bounds(build_asm_op(U32_BOUND).as_str(), U32_BOUND);
+
+    // should fail during compilation if b = 0
+    test_compilation_failure(build_asm_op(0).as_str(), "parameter");
+}
+
+#[test]
+fn u32div_full() {
+    let asm_op = "u32div.full";
+
+    // should push the quotient c = a / b onto the stack
+    // should push the remainder d = a % b onto the stack
+    test_div_full(asm_op);
+}
+
+#[test]
+fn u32div_full_fail() {
+    let asm_op = "u32div.full";
+
+    // should fail if a >= 2^32
+    test_execution_failure(asm_op, &[1, U32_BOUND], "FailedAssertion");
+
+    // should fail if b >= 2^32
+    test_execution_failure(asm_op, &[U32_BOUND, 1], "FailedAssertion");
+}
+
+#[test]
+#[should_panic = "divide by zero"]
+fn u32div_full_panic() {
+    let script = compile("begin u32div.full end");
+    let inputs = build_inputs(&[0, 1]);
+
+    // should panic if b = 0
+    execute(&script, &inputs).unwrap();
+}
```

Comment:
Similar to the other comment: this should probably return an error instead of panicking.